### PR TITLE
Ft-158998537 Request additional details on a redemption and edit a redemption

### PR DESCRIPTION
**Pull request by mkiterian**
## Description
This PR:
- Enables a CIO to request additional information for a specific redemption request
- Enables a society president to update a pending redemption request

#### CIO:
- logged in as a CIO click on the `More Info` button on a redemption request, fill in the comment section and click send
#### Society president:
- logged in as a society president click on a pending redemption card. Fill the edit form with updated info then click on edit to send

### CIO Views
screen shot 2018-07-16 at 12 16 07
request more informtion form

### Society President views
edit redemption form

### CODE REVIEW

**Comment by kimani-ndegwa** [comment]
Please resolve the conflicts here @mkiterian .

File: src/common/Select.jsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ const Select = props => (
       value={props.value}
     >
 
-      <option value={props.placeholder}>{props.placeholder}</option>
+      <option value={props.value || props.placeholder}>{props.value || props.placeholder}</option>
```
Comment (kimani-ndegwa) [inline src/common/Select.jsx]: Consider destructuring the props above so you don't need to repeat the `props.value` i.e `let {value, placeholder, options} = this.props`

File: src/components/activities/ActivityCard.jsx [hunk 2]
```diff
@@ -234,14 +281,14 @@ class ActivityCard extends Component {
...
-          <div className='activity__footer'>
-            {
-              this.renderLocationOrPoints()
-            }
-            {
-              this.renderButtonsOrStatus()
-            }
-          </div>
+        </div>
+        <div className='activity__footer'>
```
Comment (kimani-ndegwa) [inline src/components/activities/ActivityCard.jsx]: This is more of a best practice thing. Good job on delegating this functionality but what  is your opinion of calling the function before the return and then placing the necessary jsx as a variable within the return function? 

To ensure we keep the return statement as clean as possible.

File: src/constants/clickAction.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// different actions from click events
+export const APPROVE = 'approved';
```
Comment (kimani-ndegwa) [inline src/constants/clickAction.js]: One object, many keys for memory. A nit pick indeed but allows us to be conscious of how we are writing code. Also, why not create a global map that uses integers like we did for the society names that we can reuse across the whole code base?

File: src/components/activities/ActivityCard.jsx (hunk 2 above)
Comment (mkiterian) [reply src/components/activities/ActivityCard.jsx]: I don't really have strong inclinations either way. It's just a pattern that I'm used to. The JSX would look cleaner though, so I'll switch to the best practice

File: src/constants/clickAction.js (hunk 3 above)
Comment (mkiterian) [reply src/constants/clickAction.js]: @kimani-ndegwa I agree on the single object. But in terms of integer keys, I find this hard to read. Comparing `clickActions[1]` vs clickActions.APPROVE, I would go with the latter. What do you think?

File: src/containers/Redemptions.jsx [hunk 4]
```diff
@@ -220,19 +214,39 @@ class Redemptions extends React.Component {
...
+      this.setState({
+        showModal: true,
+        selectedRedemption,
+      });
+    } else if (clickAction === APPROVE) {
+      this.props.verifyRedemption(redemptionId, clickAction);
     }
   }
 
+  handleClickToEdit = (redemptionId) => {
```
Comment (Chris-Maina) [inline src/containers/Redemptions.jsx]: I noticed that the contents of this function are similar to that of `handleClick, line 223-228`,  could you shed more light on this?

Comment (mkiterian) [reply src/containers/Redemptions.jsx]: `handleClickToEdit` is called when the user clicks on the clickable area of an activity card. It sets state with the clicked on redemption by finding it using the id from all the redemptions, and also opens the modal by setting show modal to true. `handleClick` is called when the action buttons (approve, reject, comment) are clicked and calls the verification thunk

**Comment by kimani-ndegwa** [comment]
Do resolve the conflicts here @mkiterian

**Comment by kimani-ndegwa** [comment]
LGTM

**Pull request merged**